src/store/token-dashboard/slice.ts

```diff
@@ -119,8 +144,94 @@ const slice = createSlice({
 
     // Saga Actions
 
-    pressClaim: () => {},
-    pressSend: () => {}
+    pressSend: () => {},
+    fetchAssociatedWallets: () => {},
+    setAssociatedWallets: (
+      state,
+      {
+        payload: { associatedWallets }
+      }: PayloadAction<{ associatedWallets: AssociatedWallets }>
+    ) => {
+      state.associatedWallets.connectedWallets = associatedWallets
+      state.associatedWallets.confirmingWallet = null
+      state.associatedWallets.status = null
+    },
+    pressConnectWallets: state => {
+      state.modalState = {
+        stage: 'CONNECT_WALLETS',
+        flowState: { stage: 'ADD_WALLET' }
+      }
+      state.modalVisible = true
+      state.associatedWallets.removeWallet.wallet = null
+    },
+    connectNewWallet: state => {
+      state.associatedWallets.status = 'Connecting'
+    },
+    setIsConnectingWallet: (
+      state,
+      { payload: { wallet } }: PayloadAction<{ wallet: string }>
+    ) => {
+      // is connecting
+      state.associatedWallets.confirmingWallet = wallet
+    },
+    setWalletAddedConfirmed: (
+      state,
+      { payload: { wallet } }: PayloadAction<{ wallet: string }>
+    ) => {
+      state.associatedWallets.connectedWallets = (
+        state.associatedWallets.connectedWallets || []
+      ).concat(wallet)
+    },
+    setWalletRemovedConfirmed: (
+      state,
+      { payload: { wallet } }: PayloadAction<{ wallet: string }>
+    ) => {
+      state.associatedWallets.connectedWallets = (
+        state.associatedWallets.connectedWallets || []
+      ).filter(connectedWallet => connectedWallet === wallet)
```

Comment:
should the filter be NOT_EQUAL i.e. connectedWallet => connectedWallet !== wallet?

Reply:
thx, yeah I don't even use this - will delete